### PR TITLE
Add ability to customize CSS into entity configuration

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Can be customized for each entity:
image

Can also be inherited from parent configuration:
image

### CODE REVIEW

File: install/mysql/glpi-empty.sql [hunk 1]
```diff
@@ -2956,6 +2956,8 @@ CREATE TABLE `glpi_entities` (
   `date_mod` timestamp NULL DEFAULT NULL,
   `date_creation` timestamp NULL DEFAULT NULL,
   `autofill_decommission_date` varchar(255) COLLATE utf8_unicode_ci NOT NULL DEFAULT '-2',
+  `enable_custom_css` int(11) NOT NULL DEFAULT '-2',
```
Comment (trasher) [inline install/mysql/glpi-empty.sql]: why `-2`?

Comment (trasher) [reply install/mysql/glpi-empty.sql]: I guess we should use a boolean here, can't we?

Comment (cedric-anne) [reply install/mysql/glpi-empty.sql]: `-2` (https://github.com/glpi-project/glpi/blob/9.5/bugfixes/inc/entity.class.php#L56) is the value used to defines that configuration is inherited from parent. The `Entity::getUsedConfig()` ( https://github.com/glpi-project/glpi/blob/9.5/bugfixes/inc/entity.class.php#L2338 ) method will recursivly search in parents and returned the first value that is not `-2`.

Comment (trasher) [reply install/mysql/glpi-empty.sql]: ok :)

**Comment by cedric-anne** [comment]
I added usage of CodeMirror component (MIT licence) to transform the textarea into a real CSS code editor (with syntax highlighting, basic autocompletion, code folding).

image

File: install/mysql/glpi-empty.sql [hunk 2]
```diff
@@ -2956,14 +2956,16 @@ CREATE TABLE `glpi_entities` (
...
+  `custom_css_code` text COLLATE utf8_unicode_ci,
   PRIMARY KEY (`id`),
   UNIQUE KEY `unicity` (`entities_id`,`name`),
   KEY `entities_id` (`entities_id`),
   KEY `date_mod` (`date_mod`),
   KEY `date_creation` (`date_creation`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
 
-INSERT INTO `glpi_entities` VALUES ('0','Root entity','-1','Root entity',NULL,'1',NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'0',NULL,NULL,NULL,'0','0','0','0','0','0','0','0','0','0','0','-10','0','0','-10','1',NULL,'1','0','0',NULL,'0','0','0','0','0','1','-10','0','0','10','10','0','1','0',NULL,NULL,0);
+INSERT INTO `glpi_entities` VALUES ('0','Root entity','-1','Root entity',NULL,'1',NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,'0',NULL,NULL,NULL,'0','0','0','0','0','0','0','0','0','0','0','-10','0','0','-10','1',NULL,'1','0','0',NULL,'0','0','0','0','0','1','-10','0','0','10','10','0','1','0',NULL,NULL,'0','0',NULL);
```
Comment (trasher) [inline install/mysql/glpi-empty.sql]: Note to myself: to be reported on https://github.com/glpi-project/glpi/pull/5947

**Pull request merged**